test/filter.coffee

```diff
@@ -259,20 +259,20 @@ testFilter(
 			b: 'c'
 			'd'
 		]
-	"a eq ((b eq 'c') or ('d'))"
+	"a eq ((b eq 'c') or 'd')"
 )
 
 testFilter(
 	a:
 		b: 'c'
 		$eq: 'd'
-	"((a/b eq 'c') and (a eq 'd'))"
+	"(a/b eq 'c') and (a eq 'd')"
 )
 
 # Test raw strings
 testFilter(
 	$raw: "(a/b eq 'c' and a eq 'd')"
-	"(a/b eq 'c' and a eq 'd')"
+	"((a/b eq 'c' and a eq 'd'))"
```

Comment:
That’s unexpected, why is the extra set of parentheses needed now but wasn’t before?

Reply:
So strictly speaking it's not needed but because we're passed a $raw string we now need to add brackets to it to make sure it's an atomic statement as the contents could be anything, but for this particular $raw it already contains its own brackets so they get doubled up. Before when we added brackets everywhere we didn't have to do it specifically for $raw we could guarantee they would end up added elsewhere